hook.go

```diff
@@ -61,45 +93,63 @@ func (h *Hook) Levels() []logrus.Level {
 }
 
 func (h *Hook) Fire(entry *logrus.Entry) error {
-	h.q.Add(entry)
+	select {
+	case <-h.ctx.Done():
+		return fmt.Errorf("failed to write log entry: context canceled")
+	default:
+		h.entriesCh <- entry
```

Comment:
```suggestion
	case h.entriesCh <- entry:
```
here comes race bugs